src/extension/geofence/src/main/java/org/geoserver/geofence/GeofenceAccessManager.java

```diff
@@ -580,8 +661,12 @@ AccessLimits buildResourceAccessLimits(
         AccessLimits accessLimits = null;
         if (info instanceof FeatureTypeInfo) {
             // merge the area among the filters
-            if (reprojArea != null) {
-                Filter areaFilter = FF.intersects(FF.property(""), FF.literal(reprojArea));
+            if (intersectsArea != null) {
+                Filter areaFilter = FF.intersects(FF.property(""), FF.literal(intersectsArea));
```

Comment:
Ok so it seems to be already here. Why does it need to be added as a separate field too?

Reply:
thanks for your point @aaime : 
the scenario that the usage of ClipIntersectionFeatureCollection and the new field try to handle is one where we have two spatial filter one per type and is needed to put them in "OR". 
This happens when a user is inside two groups one with an intersects filter and the other with the clip. Geofence will return an accessInfo with both filter types. Simply delegating to the database the intersection and the applying the clip with the ClipFeatureCollection would return only the clipped geometries. 
In this case the intersects filter is still delegated to the database but the intersection is performed again to check wheter or not return feature according to the two spatial filters.

Reply:
So in case you have both, you need to clip the geometry, but only if it does not also overlap with the intersect filter, in which case you return it fully?
That's bizzarre, but even if we wanted to have this functionality, it should be buried inside the GeoFence module, something like this does not belong in GeoTools.

Reply:
Actually scratch that, it woudl have to be in gs-main... do we really need that behavior though?
If really needed, then the odd feature collection wrapper should be a package private class somewhere in gs-main. I mean, it's even hard to explain, we don't want to expose it.

Reply:
Odd but... if a user belongs to two groups one with an intersects filter and the other with the clip filter and the two overlaps,  if the intersects doesn't win over the clip then we would have cases where the user is meant to have access to a geometry that he cannot actually sees.